src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -809,14 +824,17 @@ def restore_backup(resource_group_name, webapp_name, storage_account_url, backup
                    target_name=None, overwrite=None, ignore_hostname_conflict=None, slot=None):
     client = web_client_factory()
     storage_blob_name = backup_name
+    webapp_info = client.web_apps.get(resource_group_name, webapp_name)
+    app_service_plan = webapp_info.server_farm_id.split('/')[-1]
     if not storage_blob_name.lower().endswith('.zip'):
         storage_blob_name += '.zip'
     location = _get_location_from_webapp(client, resource_group_name, webapp_name)
     db_setting = _create_db_setting(db_name, db_type, db_connection_string)
     restore_request = RestoreRequest(location, storage_account_url=storage_account_url,
                                      blob_name=storage_blob_name, overwrite=overwrite,
                                      site_name=target_name, databases=db_setting,
-                                     ignore_conflicting_host_names=ignore_hostname_conflict)
+                                     ignore_conflicting_host_names=ignore_hostname_conflict,
+                                     app_service_plan=app_service_plan)
     if slot:
         return client.web_apps.restore(resource_group_name, webapp_name, 0, restore_request, slot)
```

Comment:
//CC: @Nking92, i believe we should use `restore_slot`

Reply:
Fixed